ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -15713,6 +15714,7 @@ private void copyInfoToQueryProperties(QueryProperties queryProperties) {
       queryProperties.setNoScanAnalyzeCommand(qb.getParseInfo().isNoScanAnalyzeCommand());
       queryProperties.setAnalyzeRewrite(qb.isAnalyzeRewrite());
       queryProperties.setCTAS(qb.getTableDesc() != null);
+      queryProperties.setInsert(qb.getParseInfo().hasInsertTables());
```

Comment:
can we confuse this with IOW?

Reply:
okay, I just brought this from a downstream-only commit which isn't that clear...so qb.getParseInfo().hasInsertTables() is true in all these cases:
```
INSERT [OVERWRITE], UPDATE, DELETE, MERGE
```
that's because all the UPDATE/DELETE/MERGE operations are rewritten to INSERTs, so isInsert contains much confusion here, so...hm, let me rewrite this to isDML, because that's the point here I want to exactly catch, if someone needs isInsert for only INSERT INTO queries, it can be implemented later, I don't have to bother with this now I think